eppo/src/main/java/cloud/eppo/android/ConfigurationRequestor.java

```diff
@@ -36,27 +43,39 @@ public void onCacheLoadSuccess() {
 
             @Override
             public void onCacheLoadFail() {
-                // no-op; fall-back to Fetch
+                cacheLoadInProgress.set(false);
+                Log.d(TAG, "Did not initialize from cache");
+                // If cache loading failed, and fetching failed, fire off the failure callback if not yet done so
+                // Otherwise, if fetching has not failed yet, defer to it for firing off callbacks
```

Comment:
What happens is about the same time the fetching exceptional handler checks `cacheLoadInProgress`, thinks it is in progress and returns? I guess this is covered right - we will eventually either get good data from the cache or empty data from the cache (if its empty). This is a good endpoint.

Reply:
Yeah so if `cacheLoadInProgress` is checked right before it's set to false, then the fetcher won't fire off the callbacks but the cache load completion will because the final check for callback called flag is an atomic check and a set